In [36]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
from scipy.stats import mannwhitneyu

In [37]:
df = pd.read_csv('source/FReDA4.csv')

In [38]:
couples_agreement = df[df["Group2"] == "Couple Agreement"].copy()
couples_deprived = df[df["Group2"] == "Couple Deprivation"].copy()
couples_saturated = df[df["Group2"] == "Couple Saturation"].copy()

subjects_deprived = df[df["Group1"] == "SubGroup2"].copy()
subjects_saturated = df[df["Group1"] == "SubGroup5"].copy()

In [39]:
couples = len(couples_deprived)
married = (couples_deprived["Married"] == 1).sum()
kids = (couples_deprived["Kids"] >= 1).sum()
together = (couples_deprived["Cohabitation"] >= 1).sum()
orientation = (couples_deprived["Relationship Sex"] >= 1).sum()
work = (couples_deprived["Work Status"] >= 1).sum()

per_married = married*100/couples
per_kids = kids*100/couples
per_together = together*100/couples
per_orientation = orientation*100/couples
per_work = work*100/couples

In [40]:
none = df.loc[df["Group2"] == "Couple Saturation", "Relationship Length"].isna().sum()
pd.crosstab(
    couples_deprived["Relationship Length"],
    couples_deprived["Married"],
)

Married              0.0   1.0
Relationship Length           
1.0                  182    14
2.0                  262   140
3.0                  302   626
4.0                  130  1188
5.0                   20   484

In [41]:
pd.crosstab(
    subjects_deprived["Kids"],
    subjects_deprived["Sex"],
)

Sex   0.0  1.0
Kids          
0.0   413  384
1.0   224  236
2.0   302  304
3.0   107  110

In [42]:
pd.crosstab(
    subjects_saturated["Kids"],
    subjects_saturated["Sex"],
)

Sex   0.0  1.0
Kids          
0.0    74   76
1.0    24   28
2.0    29   31
3.0    10   10

Satisfied couples are in mayority heterosexuals (96.5%) and live together (89.3%). Most of them works outside home either part of fulltime (91.0%).
More than half of them are married (57.7%), while less than half have Kids (44.5%)
Married couples are together for longer time, 88% of them has been together more than Five years. Only 32% of single couples has been together more than Five years.
Married couples also are the most prominents to have childre, where 66% Reports of having one or more than one kid in the Household.

Couples tend to be evenly regarding age difference, where 38% of single couples have the same age and 33% of married has the same age.


Deprived couples are again in mayority heterosexuals (97,3%) and live together (93.4%). Again most of them works outside their homes (85,2%)
More than 2/3 are married (72.4%), while nearly the same amount have Kids (74%).
Married couples are together for longer time, 94% of them has been together more than Five years. Single couples instead are distributed evenly regarding relationship length. However, most of them have together less than 10 years (83%), while married ones are mostly more than that time (68%)
Kids are mostly within married couples in this Group, 87%. Contrary, single couples prefers not to have childer (60%)

Saturated couples are mostly heterosexuals (95.6%) and live together (97.7%). Most of the work outside home (83.3%)
68% are married, and less than half have Kids (48.8%)
Married couples are together for longer time (83%) are together more than 5 years. However, only 38% of them are together for more than 10 years.
Single couples are together mostly below 5 years (64.28%)
85 of single couples does not have Kids, while 64% of married couples have at least 1